packages/react/src/provider/use-open-feature-client-status.ts

```diff
@@ -28,6 +33,12 @@ export function useOpenFeatureClientStatus(): ProviderStatus {
       client.removeHandler(ProviderEvents.Reconciling, updateStatus);
     };
   }, [client]);
+  // highest priority > evaluation hook options > provider options > default options > lowest priority
+  const defaultedOptions = { ...DEFAULT_OPTIONS, ...useProviderOptions(), ...normalizeOptions(options) };
+
+  if (defaultedOptions.suspendUntilReady && status === ProviderStatus.NOT_READY) {
+    suspendUntilReady(client);
```

Comment:
Some of the code here is identical to the `useWhenProviderReady` hook - we might want to commonize it into a util.